cdap-securestore-spi/src/main/java/co/cask/cdap/securestore/spi/SecretAlreadyExistsException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+/**
+ * Exception that indicates a secret already exists.
+ */
+public class SecretAlreadyExistsException extends IllegalArgumentException {
+  public SecretAlreadyExistsException(String message) {
+    super(message);
```

Comment:
Should have a constructor which takes a Secret and generate the required message. Look at DatasetNotFoundException and DatasetAlreadyExistsException for examples.

Reply:
Secret shouldn't be part of the message in any case. It can take the name, but never the value.

Reply:
I think taking a message provides flexibility on implementation side to throw exception with appropriate message rather than fixating it in the spi. What do you think @rohitsinha54 @chtyim

Reply:
It would be good to have a constructor that enforces a fixed error message format such that different SPI implementations doesn't need to care about how to format the message to be consistent.

Reply:
Ok, added